### DISCUSSION TITLE
Graphics card AMD HD 4350/4550 + Debian 12/"Bookworm" + LxQt from source code: Screen continuously turns black and shows different resolutions, all this in an endless loop

### DISCUSSION THREAD

**Discussion by ghost** [lxqt/lxqt#2491] [Q&A] [answered]
AMD Radeon HD 4350/4550 and Debian 12 (currently stable Debian release): 

Once in a while the screen turns black for some seconds, then the LxQt GUI gets displayed in another resolution for a while, then the screen turns black again for some seconds, the the LxQT GUI again gets displayed in another resolution, all this in an endless loop.

Every time before the screen turns black, a message box gets displayed in the lower right corner of the LxQt GUI. This message box displays the following information:

> lxqt-config-monitor: Standardbildschirmeistellungen wurden angewendet. Wenn Sie die Bildschirmeinstellungen ändern möchten, verwenden Sie bitte lxqt-config-monitor.

Translated by myself to English:

> lxqt-config-monitor: Default screen settings have been applied.  If you want to change the screen settings please use lxqt-config-monitor.

These are the screen resolutions that were displayed one after another during one LxQt session which lasted about 15 minutes:

1. 1024 x 768
2. 1920 x 1080
3. 1024 x 768
4. 1920 x 1080

This behavior can be reduced to the screen only turning black for some seconds once in a while without then changing the screen resolution each time, plus showing lxqt-config-monitor message box once in a while, if one uses an xorg.conf file in /etc/X11: Then the screen resolution keeps to the resolution defined in the xorg.conf file.

The behavior described above occurs after nearly every single login into the LxQt GUI, also after reboots.

I implemented LxQt 1.3.0 on the affected machine by compiling its source files which I checked out from gitlab.com using git. Before the implementation of LxQt 1.3.0 I had deleted the LxQt deb packages for LxQt version 1.2.0 - in this LxQt version the behavior as described above occurred, too.

LxQt version on the affected machine, information copied from within the "About LxQt" box:

```
LXQt Desktop Toolbox - Technical Info
  LXQt About Version:     1.3.0
  LXQt Version:           1.3.0
  Qt:                     5.15.8
  Build type:             Release
  System Configuration:   /etc/xdg
  Share Directory:        /usr/share/lxqt
  Translations:           /home/xxx/.local/share/lxqt/translations
/usr/local/share/lxqt/translations
/usr/share/lxqt/translations

User Directories
  Xdg Data Home:          /home/xx/.local/share
  Xdg Config Home:        /home/xx/.config
  Xdg Data Dirs:          /home/xxx/.local/share
/usr/local/share
/usr/share
  Xdg Cache Home:         /home/xxx/.cache
  Xdg Runtime Home:       /run/user/1000
  Xdg Autostart Dirs:     /etc/autostart
/etc/xdg/autostart
/usr/share/autostart
  Xdg Autostart Home:     /home/xxx/.config/autostart

```
Debian version on the affected machine:

```
xxx@yyy:~$ cat /etc/debian_version 
12.1
xxx@yyy:~$ 
```

Running kernel version, currently the only one installed kernel version:

```
xxx@yyy:~$ uname -a
Linux yyy 6.1.0-11-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.38-4 (2023-08-08) x86_64 GNU/Linux
xxx@yyy:~$
```

Information about the graphics card in the affected machine:

```
root@syyy:~# lshw -C video
  *-display                 
       description: VGA compatible controller
       product: RV710 [Radeon HD 4350/4550]
       vendor: Advanced Micro Devices, Inc. [AMD/ATI]
       physical id: 0
       bus info: pci@0000:01:00.0
       logical name: /dev/fb0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: pm pciexpress msi vga_controller bus_master cap_list rom fb
       configuration: depth=32 driver=radeon latency=0 resolution=1920,1080
       resources: irq:25 memory:d0000000-dfffffff memory:fdee0000-fdeeffff ioport:ee00(size=256) memory:c0000-dffff
root@yyy: 
```

Content of the xorg.conf file in /etc/X11:

```
Section "Monitor"
       Identifier "VGA-0"
       Modeline "1920x1080R" 138.50 1920 1968 2000 2080 1080 1083 1088 1111 +hsync -vsync
       Option "PreferredMode" "1920x1080R"
EndSection

```
I created the content of the xorg.conf file with the outputs of the commands "xrandr" and "cvt -r <resolution>".

Xfce:
=====

The desktop environment Xfce 4.18, i. e. its deb packages, is installed on the affected machine and can be used besides the compiled and implemented LxQt version 1.3.0 from source.

Installed deb packages for GTK on this machine in version 3.24.37.

The behavior as described above does **not** occur in Xfce on the affected machine. I checked this during several Xfce sessions which each lasted for at least 10 minutes, most of them initialized by a reboot. Only rarely the screen flickers for about a second, that's all. I also wrote this bug report in one of these Xfce sessions.

(L)Ubuntu and other LxQt versions (not) being affected:
==========================================

The behavior as described above occurred on the same machine in Lubuntu 22.04.3 LTS with each the 6.2.0-26-generic kernel and the 5.15.0-79-generic kernel running, too. According to https://lubuntu.me/jammy-3-released/, LxQt 0.17.0 is part of Lubuntu 22.04.

The behavior did *not* occur in Lubuntu 20.04 LTS. I curreńtly cannot find out the LxQt version being part of Lubuntu 22.04. LTS

Both Lubuntu 20.04 and 22.04.3 are not installed anymore on the affected machine.

For more details about all this see my thread in the Lubuntu forum at

https://discourse.lubuntu.me/t/graphic-card-amd-radeon-hd-4350-4550-and-lubuntu-22-04-3-screen-constantly-turns-black-for-some-seconds/4490/7

Conclusion:
=========

The cause(s) for the behavior described in this thread must be the respective LxQT version installed as deb packages from the official Debian repositories of Debian 12, the installed deb packages from the official repositories of Lubuntu 22.04, **and LxQt version 1.3.0 built from source code**, maybe in connection with the 6.* kernel versions running in Debian 12 and Lubuntu 22.04.

**Comment by lorn10** [comment, marked as answer]
Hi there, question did your try also a more recent **Mesa devel** version? I would recommend to test the [oibaf PPA](https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers) if you have Lubuntu installed.

In recent weeks and months there was some (super) heavy code refactoring on the Mesa **r600** driver. So I think it would not harm to test a newer Mesa devel release. :wink: I am currently on `Mesa 24.0~git2311050600.cd0a01~oibaf~j (git-cd0a015 2023-11-05 jammy-oibaf-ppa)` and it works fine on all my Radeon Hardware down to my Radeon HD 2400 XT GPU. 

However, this is true for KDE and Kubuntu but I plan to migrate at least my weakest hardware to LXQT and Lubuntu.

And finnaly, exist there a way to set the renderer in LXQT to **OpenGL ES** & **EGL** mode? This is possible in KWin on Kubuntu and I am looking for a similar option in LXQT. This can [really help](https://bugs.kde.org/show_bug.cgi?id=442386) on older hardware and for me it makes Kubuntu usable on my old iMac 5,1 where the normal OGL mode (in conjunction with GLX) seems to be broken.

**Reply by ghost** [reply]
I marked this comment as "answer" because I found out that an update of Mesa-related packages in Debian 12 (stable, "Bookworm") must have resolved the behavior as described in this thread, as suggested by @lorn10 - thank you for your hint ;-)

But here is an advice for all possible future readers of this thread searching for a solution because their machines are affected by the same or a similar behavior as described in this thread: 

I would not immediately use the above-mentioned oibaf PPA in the first place, mentioned above by @lorn10: 

To keep the integrity of your package management system on your machine I would first install updates (if available) for all Mesa-related packages using the package management system on your machine. Only if that does not resolve the behavior as described in this thread, I would make use of the above-mentioned oibaf PPA.

**Comment by tsujan** [comment]
> This is possible in KWin on Kubuntu 

When it's possible in KWin, it's also possible when KWin is used anywhere, whether in KDE or in LXQt. LXQt doesn't do these things; the compositor you use with LXQt does.

I'm converting this to a discussion....

**Reply by ghost** [reply]
An update:

I did not start the affected machine since about two months. Today I started it again because this thread here received the first replies yesterday. 

So I updated the Debian packages which are installed on the affected machine by package updates from the Debian repositories. These updates included among other things a kernel update. 

This is the currently installed kernel version on the affected machine:

```
xxx@yyy:~$ uname -a
Linux yyy 6.1.0-13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.55-1 (2023-09-29) x86_64 GNU/Linux
xxx@yyy:~$
```
Current Debian version:

```
xxx@yyy:~$ cat /etc/debian_version 
12.2
xxx@yyy:~$
```

This is an information for @lorn10 (thank you for your hints, but I did not use the Mesa-related PPA which you suggested), this is a list of all Debian Bookworm packages currently installed on the affected machine with the pattern "mesa" in their package names:

```
xxx@yyy:~# apt list *mesa* --installed
Auflistung… Fertig
libegl-mesa0/stable,now 22.3.6-1+deb12u1 amd64  [Installiert,automatisch]
libgl1-mesa-dri/stable,now 22.3.6-1+deb12u1 amd64  [installiert]
libglapi-mesa/stable,now 22.3.6-1+deb12u1 amd64  [Installiert,automatisch]
libglu1-mesa-dev/stable,now 9.0.2-1.1 amd64  [Installiert,automatisch]
libglu1-mesa/stable,now 9.0.2-1.1 amd64  [Installiert,automatisch]
libglx-mesa0/stable,now 22.3.6-1+deb12u1 amd64  [installiert]
mesa-utils-bin/stable,now 8.5.0-1 amd64  [Installiert,automatisch]
mesa-utils/stable,now 8.5.0-1 amd64  [installiert]
mesa-va-drivers/stable,now 22.3.6-1+deb12u1 amd64  [Installiert,automatisch]
mesa-vdpau-drivers/stable,now 22.3.6-1+deb12u1 amd64  [Installiert,automatisch]
mesa-vulkan-drivers/stable,now 22.3.6-1+deb12u1 amd64  [installiert]
xxx@yyy:~# 
```

The affected machine is now running about 45 minutes displaying the LxQT desktop environment - and since the beginning of this LxQT session I cannot notice the behavior which I described in this thread in my first post.

So this issue can be considered as solved.

@tsujan, I would like to ask you to turn this discussion back again into an issue, with the status "solved" or "closed" or something like that. 

Because I had been confronted with a software issue which I reported here, invested several hours to narrow the possible causes of that software issue, wrote the initial post in this thread which took me a considerable amount of time (my mother tongue is not English), and I finally found the solution for the software issue I had been reporting here in this thread.

**Reply by lorn10** [reply]
Great so it looks that this was in the end an issue in Mesa which was resolved by an update of Mesa. :tada: 

And by the way while Mesa 22.3.6 is much newer than the previously used one it is still clearly behind Mesa 24.x. So if someone is experiencing similar problems with older Radeon hardware it is strongly recommended to upgrade to a recent Mesa devel release. Even if the r600 driver is old it was as mentioned radically refactored in recent time (including the total conversion to NIR and purging of the old TGSI stuff.)

**Reply by tsujan** [reply]
> I would like to ask you to turn this discussion back again into an issue, with the status "solved" or "closed" or something like that.

https://github.com/lxqt/lxqt/issues is LXQt's bug tracker, while your problem wasn't caused by LXQt; hence converting it to a discussion, from which other users might benefit.

Here you could mark @lorn10's comment as the answer (see the button below the comment), to show that the problem is solved and how it is solved.

**Reply by tsujan** [reply]
@lorn10
I think you could do it too. I don't have Radeon.
